abdiff/core/run_ab_transforms.py

```diff
@@ -43,63 +82,57 @@ def run_ab_transforms(
         f"{[transformed_directory_a, transformed_directory_b]}"
     )
 
-    # create list of run configs
     run_configs = [
         (image_name_a, transformed_directory_a),
         (image_name_b, transformed_directory_b),
     ]
 
-    # transform input files using images A and B via Docker containers
-    # save transformed record JSON files in corresponding directory
     containers = []
     for input_file in input_files:
+        source, output_file = parse_transform_details_from_extract_filename(input_file)
         for docker_image, transformed_directory in run_configs:
             container = run_docker_container(
-                docker_image, transformed_directory, input_file, docker_client
+                docker_image,
+                transformed_directory,
+                source,
+                input_file,
+                output_file,
+                docker_client,
             )
             containers.append(container)
 
-    # wait for all container runs to complete (status = "exited")
     wait_for_containers(containers, timeout)
     logger.info(f"All {len(containers)} containers have exited.")
 
-    # get logs from all Docker containers
-    log_file = f"{run_directory}/transformed/logs.txt"
-    with open(Path(log_file), "w") as file:
-        for container in containers:
-            header = (
-                f"docker_image: {container.labels['docker_image']} | "
-                f"source: {container.labels['source']} | "
-                f"input_file: {container.labels['input_file']}\n"
-            )
-            file.write(header)
-            file.write(container.logs().decode())
-            file.write("\n\n")
+    log_file = aggregate_logs(run_directory, containers)
```

Comment:
Commenting here arbitrarily, but thanks for the updates around here.  Looking at the full code locally, it's easy to scan and understand.